### PR TITLE
SAA Membership Card Test

### PR DESCRIPTION
**Pull request by William-Misiaszek**
# NOT READY FOR REVIEW
- (Edit the above to reflect status)

# Summary
- TL;DR - what's this PR for?

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Navigate to...
3. Verify...

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -63,3 +63,8 @@ yarn-error.log
 
 # Local Netlify folder
 .netlify
+
+src/stanford.pass
+certs
+src/saacard.pass/pass.json
+/src/netlify/certs
```
Comment (sherakama) [inline .gitignore:70]: @William-Misiaszek 

Hi. Please let me know if you'd like me to add any certs, keys, tokens, or other credentials to the vault.

File: localhost-key.pem [hunk 2]
```diff
@@ -0,0 +1,28 @@
+-----BEGIN PRIVATE KEY-----
```
Comment (sherakama) [inline localhost-key.pem:1]: Please delete this file and use the localhost keys that are already in `/dev`

File: netlify.toml [hunk 3]
```diff
@@ -2,11 +2,15 @@
 # ##############################################################################
 [build]
   publish = "public"
+  functions = "src/netlify/functions"
```
Comment (sherakama) [inline netlify.toml:5]: Please use [Gatsby's functions](https://www.gatsbyjs.com/docs/reference/functions/routing/) over directly writing a Netlify Function. Netlify's deploy plugin will automatically package up and create Netlify functions from Gatsby functions.

File: src/netlify/functions/generate-pass.js
Comment (sherakama) [inline src/netlify/functions/generate-pass.js:1]: Please convert this file into a Gatsby Function vs a direct Netlify function.

Comment (William-Misiaszek) [reply src/netlify/functions/generate-pass.js:1]: Converted the function to a Gatsby function in src/api/applewalletcard/generate-pass.js. Removed the toml configuration too.

File: netlify.toml (hunk 3 above)
Comment (William-Misiaszek) [reply netlify.toml]: Removed the netlify function config line.

File: localhost-key.pem (hunk 2 above)
Comment (William-Misiaszek) [reply localhost-key.pem]: Deleted!

File: .gitignore (hunk 1 above)
Comment (William-Misiaszek) [reply .gitignore:70]: Will message you the updated certs in Slack!

File: localhost-key.pem (hunk 2 above)
Comment (sherakama) [reply localhost-key.pem]: TY

File: netlify.toml (hunk 3 above)
Comment (sherakama) [reply netlify.toml]: Ty

File: src/netlify/functions/generate-pass.js
Comment (sherakama) [reply src/netlify/functions/generate-pass.js:1]: ty

**Comment by William-Misiaszek** [comment]
I merged "dev" into this branch to get our latest version of the site. Working through the checks/integration checks. 

Goal is to confirm the pass file is served by pressing the Add to Apple Wallet button.

File: src/api/walletpass/apple.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { generateAplleWalletPass } from '../../utilities/walletPass';
+
+export default async function handler(req, res) {
+  try {
+    // TODO: Get the current user information:
+    // - firstName
+    // - lastName
+    // - membershipNumber
```
Comment (moisesnarvaez) [inline src/api/walletpass/apple.js:8]: @sherakama Hi! 👋 . Can you help us know how to get the current user information in the API? We need this information to generate the Apple pass dynamically.

CC: @William-Misiaszek

Comment (sherakama) [reply src/api/walletpass/apple.js:8]: Hi @moisesnarvaez,

Thanks for reaching out. Here are a couple of resources for you.

API Client: https://github.com/SU-SWS/saa_alumni/blob/dev/src/utilities/MegaProfile.js

You'll need the authenticated user's encodedSUID value to make the request and you can get that from the `user` object after you authenticate the user with the Auth connect middleware. 

See: https://github.com/SU-SWS/saa_alumni/blob/dev/src/api/private-proxy.js#L45

And then https://github.com/SU-SWS/saa_alumni/blob/dev/src/api/private-proxy.js#L30-L32

The endpoints for the information you need are the `/fullprofile` and `/memberships` endpoints

See: https://github.com/SU-SWS/saa_alumni/blob/dev/src/api/auth/profile.js#L25-L28

Let me know if this is enough to get you started or if you need more information.

Comment (moisesnarvaez) [reply src/api/walletpass/apple.js]: @sherakama Thanks for the great explanation!

I did the implementation, but the middleware in the API is not working as expected:

https://github.com/SU-SWS/saa_alumni/pull/782/files#diff-c136fdbf4f69b75b172fdbaefe8b561345ef505dd00029a5f64e3def27dcaf1aR57

It's returning this error:
image
--- 
PD: The session seems to be working fine in local environment:
image

Comment (sherakama) [reply src/api/walletpass/apple.js]: @moisesnarvaez 

You'll need to have an active and valid session to access the API endpoint after implementing the auth middleware. 

The user you see in the `/editor` route is a 'mock' user as that path is authorized with the storyblok API key. I have removed the image from this post as it contains a sensitive key. Please do not share those URLs publicly as that storyblok key is a preview key and could lead to users getting access to unpublished content.  

The easiest path forward for you would be to log in by clicking the login link at the top right of the UI. You should be redirected to log in through a UAT auth service. Do you have a Stanford Pass account? You should be able to log in as yourself on the UAT environment as the users and data are sync'd back from production. If you don't have an account you could try and register one at https://uat--adapt-myaccount.netlify.app/register.

Comment (moisesnarvaez) [reply src/api/walletpass/apple.js]: Thanks for guiding me, I had to hack manually the isStoryblokEditor function (not committed) to display the real user information

PD: I'm sorry about the image.

**Comment by moisesnarvaez** [comment]
These changes were included in: https://github.com/SU-SWS/saa_alumni/pull/833/

**Comment by William-Misiaszek** [comment]
Increased scope to add Google Wallet to our initial release. We are combining Apple + Android into the [this PR](https://fullstacklabs.slack.com/archives/D04LUBLBF9B/p1713719064950149)

**Pull request closed without merging**